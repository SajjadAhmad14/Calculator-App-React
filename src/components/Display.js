import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Display extends Component {
  render() {
    return <div className = 'display-board'>
    <h1 className = 'result'>{this.props.result }</h1>
    </div>
  }
}

  Display.propTypes = { 
    result: PropTypes.string 
  }

  Display.defaultProps = {
    result: 0,
  }