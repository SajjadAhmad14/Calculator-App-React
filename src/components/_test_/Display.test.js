import React from 'react-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from '../Display';

describe('Display component', () => {
  it('Display 0 as a result', () => {
    render(<Display result = '0' />);
    const text = screen.getByText('0');
    expect(text).toMatchSnapshot();
  })

  it('Display undefined as a result', () => {
    render(<Display result = 'undefined' />);
    const text = screen.getByText('undefined');
    expect(text).toMatchSnapshot();
  })

  it('Display null as a result', () => {
    render(<Display result = 'null' />);
    const text = screen.getByText('null');
    expect(text).toMatchSnapshot();
  })
})