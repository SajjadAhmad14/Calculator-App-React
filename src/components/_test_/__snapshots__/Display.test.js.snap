// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component Display 0 as a result 1`] = `
<h1
  class="result"
>
  0
</h1>
`;

exports[`Display component Display null as a result 1`] = `
<h1
  class="result"
>
  null
</h1>
`;

exports[`Display component Display undefined as a result 1`] = `
<h1
  class="result"
>
  undefined
</h1>
`;
